packages/ag-charts-community/src/chart/data/dataDomain.test.ts

```diff
@@ -0,0 +1,552 @@
+import { describe, expect, it } from '@jest/globals';
+
+import { BandedDomain, ContinuousDomain, DiscreteDomain } from './dataDomain';
+
+describe('DiscreteDomain', () => {
+    describe('default Set-based mode', () => {
+        it('should store and retrieve values using Sets', () => {
+            const domain = new DiscreteDomain();
+            domain.extend('a');
+            domain.extend('b');
+            domain.extend('c');
+
+            expect(domain.getDomain()).toEqual(['a', 'b', 'c']);
+            expect(domain.isSortedUniqueMode()).toBe(false);
+        });
+
+        it('should deduplicate values in Set mode', () => {
+            const domain = new DiscreteDomain();
+            domain.extend('a');
+            domain.extend('b');
+            domain.extend('a'); // duplicate
+            domain.extend('c');
+            domain.extend('b'); // duplicate
+
+            const result = domain.getDomain();
+            expect(result.length).toBe(3);
+            expect(new Set(result)).toEqual(new Set(['a', 'b', 'c']));
+        });
+
+        it('should handle Date values by converting to timestamps', () => {
```

Comment:
Does it store dates as timestamps or dates internally? The test name implies it stores them as timestamps but they are retrieved as dates in the test?